plugins/CoreAdminHome/Commands/InvalidateReportData.php

```diff
@@ -68,7 +71,20 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $sites = $this->getSitesToInvalidateFor($input);
         $periodTypes = $this->getPeriodTypesToInvalidateFor($input);
         $dateRanges = $this->getDateRangesToInvalidateFor($input);
-        $segments = $this->getSegmentsToInvalidateFor($input, $sites);
+
+        $segments = array();
+        if ($input->getOption('segment')) {
+            $segments = $this->getSegmentsToInvalidateFor($input, $sites);
+        }
+        if ($input->getOption('idsegments')) {
+            $idsegments = $this->getSegmentsIdToInvalidateFor($input, $sites);
+        } else {
+            $segments = $this->getSegmentsToInvalidateFor($input, $sites);
+        }
+        if (isset($idsegments)) {
+            $segments = array_merge($segments, $idsegments);
+            $segments = array_unique($segments);
+        }
```

Comment:
I think the code here can be simplified. `$segments = $this->getSegmentsToInvalidateFor($input, $sites);` is executed no matter what but can be called twice if `--segment` is supplied. And the `if (isset($idsegments)) {` block could be located next to `$idsegments` w/o the `if`. Can you make these changes?